test/integration/acceptance/grant_permission_test.cpp

```diff
@@ -0,0 +1,884 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include "acceptance_fixture.hpp"
+#include "builders/protobuf/transaction.hpp"
+#include "builders/protobuf/queries.hpp"
+#include "framework/integration_framework/integration_test_framework.hpp"
+#include "framework/specified_visitor.hpp"
+#include "validators/permissions.hpp"
+
+using namespace integration_framework;
+using namespace shared_model;
+
+namespace shared_model {
+  namespace proto {
+    class QueryResponse;
+  }
+}
+
+class GrantPermissionTest : public AcceptanceFixture {
+ public:
+
+  /**
+   * Creates the transaction with the user creation commands
+   * @param perms are the permissions of the user
+   * @return built tx and a hash of its payload
+   */
+  auto makeAccountWithPerms(const std::string &user,
+                            const crypto::Keypair &key,
+                            const std::vector<std::string> &perms,
+                            const std::string &role) {
+    return createUserWithPerms(user, key.publicKey(), role, perms)
+        .build()
+        .signAndAddSignature(kAdminKeypair)
+        .finish();
+  }
+
+  /**
+   * Forms a transaction such that creator of transaction
+   * grants permittee a permission
+   * @param creatorAccountName — a name, e.g. user
+   * @param permitteeAccountId — an id of grantee, mostlikely name@test
+   * @param grantPermission — any permission from the set of grantable
+   * @return a Transaction object
+   */
+  proto::Transaction accountGrantToAccount(const std::string &creator_account_name,
+                                           const crypto::Keypair &creator_key,
+                                           const std::string &permitee_account_name,
+                                           const std::string &grant_permission) {
+    const std::string creator_account_id = creator_account_name + "@" + kDomain;
+    const std::string
+        permitee_account_id = permitee_account_name + "@" + kDomain;
+    return TestUnsignedTransactionBuilder()
```

Comment:
I think `baseTx()` can be reused from AcceptanceFixture
similar to the following functions

Reply:
ok

Reply:
As far as I've understand when baseTx() is used you cannot modify creator account id. Can you check it or help me?

Reply:
Indeed. But is that really necessary to set particular user?

Reply:
I am using two separate accounts, so the interface of baseTx() is not really useful in this case. Maybe for other fixtures this is helpful